### ISSUE TITLE
VRF model does not correctly calculate COP when cycling below min PLR

### ISSUE THREAD

**Issue by rraustad** [NREL/EnergyPlus#7628]
Issue overview
--------------
Recent testing has shown that the COP calculation is incorrect when operating below min PLR.

In function CalcVRFCondenser the total cooling/heating capacity is adjusted by operating PLR. Additionally, if the system is cycling below Min PLR there is a calculation for cycling ratio which is not included here. This means the COP calculation is incorrect below min PLR.

       VRF(VRFCond).TotalCoolingCapacity = TotalCondCoolingCapacity * CoolingPLR;
       VRF(VRFCond).TotalHeatingCapacity = TotalCondHeatingCapacity * HeatingPLR;

As a test, these lines were moved below the calculation for CyclingRatio and added to the equation. Further investigation showed that the local variables CoolingPLR and HeatingPLR also needed to be updated when VRF(VRFCond).VRFCondPLR is capped at MinPLR.

        if (VRF(VRFCond).MinPLR > 0.0) {
            CyclingRatio = min(1.0, VRF(VRFCond).VRFCondPLR / VRF(VRFCond).MinPLR);
            if (VRF(VRFCond).VRFCondPLR < VRF(VRFCond).MinPLR && VRF(VRFCond).VRFCondPLR > 0.0) {
                VRF(VRFCond).VRFCondPLR = VRF(VRFCond).MinPLR;
                if (CoolingPLR > 0.0) CoolingPLR = VRF(VRFCond).MinPLR; // these also need updating
                if (HeatingPLR > 0.0) HeatingPLR = VRF(VRFCond).MinPLR;
            }
        }
        VRF(VRFCond).VRFCondCyclingRatio = CyclingRatio; // report variable for cycling rate
        VRF(VRFCond).TotalCoolingCapacity = TotalCondCoolingCapacity * CoolingPLR * CyclingRatio;
        VRF(VRFCond).TotalHeatingCapacity = TotalCondHeatingCapacity * HeatingPLR * CyclingRatio;

Also noticed in this testing was that the calculations for heat recovery are above these lines of code. This means the performance of heat recovery will also be affected since the PLR used in the performance curves has not been capped at the MinPLR yet. I.e., going in to this section of code.

        VRF(VRFCond).VRFCondPLR = max(CoolingPLR, HeatingPLR);

        HRHeatRequestFlag = any(TerminalUnitList(TUListNum).HRHeatRequest);
        HRCoolRequestFlag = any(TerminalUnitList(TUListNum).HRCoolRequest);
        HREIRFTConst = 1.0;
        Real64 HREIRAdjustment = 1.0;

        if (!DoingSizing && !WarmupFlag) {
            if (HRHeatRequestFlag && HRCoolRequestFlag) {
                a calculation below here uses VRF(VRFCond).VRFCondPLR

Since VRF(VRFCond).VRFCondPLR is used inside the heat recovery code, it is necessary to cap the operating PLR above the heat recovery code (shown here) and use this local variable inside the heat recovery code where performance curves are evaluated.

    VRF(VRFCond).VRFCondPLR = max(CoolingPLR, HeatingPLR);
    Real64 tmpVRFCondPLR = 0.0;
    if (CoolingPLR > 0.0 || HeatingPLR > 0.0) tmpVRFCondPLR = max(VRF(VRFCond).MinPLR, VRF(VRFCond).VRFCondPLR);

Defect file result:

![image](https://user-images.githubusercontent.com/5778156/69274297-12709200-0ba8-11ea-95d2-4de11eb35e33.png)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added ([DevSupport\7628](https://github.com/NREL/EnergyPlusDevSupport/tree/master/DefectFiles/7000s/7628))
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
@rraustad @Nigusse When I run this defect file with develop and with the #7701 branch, I get identical results. How do I see the impact of the fix?

**Comment by rraustad** [comment]
I think because it was a Trane defect file and I must not have found the correct file. The symptom was that when power and capacity were added up that calculation did not match the reported COP. I worked weeks to figure that out.

From that issue description:

*VRF heating COPs found to be unusually low during TRANE study. Was found that the cycling ratio term was not being properly handled in COP and current total heating and cooling capacity calculations. @rraustad found coding changes to correct the COP reporting. Affects times when PLR is at the minimum PLR and the outdoor unit is cycling on and off.*
